File: ios/EdgeCoreWebViewManager.m
#import <React/RCTUIManager.h>

@interface RCT_EXTERN_MODULE(EdgeCoreWebViewManager, RCTViewManager)
RCT_EXPORT_VIEW_PROPERTY(onMessage, RCTDirectEventBlock)
RCT_EXPORT_VIEW_PROPERTY(onScriptError, RCTDirectEventBlock)
RCT_EXPORT_VIEW_PROPERTY(allowDebugging, BOOL)
RCT_EXPORT_VIEW_PROPERTY(source, NSString)
RCT_EXTERN_METHOD(runJs:(nonnull NSNumber *)reactTag js:(nonnull NSString *)js)
@end

File: ios/EdgeCoreWebViewManager.swift
@objc(EdgeCoreWebViewManager) class EdgeCoreWebViewManager: RCTViewManager {
  override static func requiresMainQueueSetup() -> Bool { return false }
  override func view() -> UIView! { return EdgeCoreWebView() }

  @objc func runJs(_ reactTag: NSNumber, js: String) {
    bridge.uiManager.addUIBlock({ (uiManager, viewRegistry) in
      let view = uiManager?.view(forReactTag: reactTag)
      if let webView = view as? EdgeCoreWebView {
        webView.runJs(js: js)
      }
    })
  }
}
